dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/dragen/DragenWorkflowAnalysis.java

```diff
@@ -0,0 +1,76 @@
+package ca.on.oicr.gsi.runscanner.dto.dragen;
+
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+
+public class DragenWorkflowAnalysis {
+  private List<DragenAnalysisUnit> analyses = new LinkedList<>();
+  private Instant completionTime;
+  private Instant startTime;
+
+  public DragenAnalysisUnit get(String sample, String lane, String index1, String index2) {
+    return get(sample, lane, new StringBuilder(index1).append("-").append(index2).toString());
+  }
+
+  public DragenAnalysisUnit get(String sample, int lane, String index1, String index2) {
+    return get(sample, lane, new StringBuilder(index1).append("-").append(index2).toString());
+  }
+
+  public DragenAnalysisUnit get(String sample, String lane, String index) {
+    return get(sample, Integer.parseInt(lane), index);
+  }
+
+  public DragenAnalysisUnit get(String sample, int lane, String index) {
+    return analyses
+        .stream()
+        .filter(a -> a.getSample().equals(sample) && a.getLane() == lane)
+        .findFirst()
+        .orElse(new DragenAnalysisUnit());
+  }
```

Comment:
This is a "get" method, but it also creates something new, empty, and invalid if the thing requested doesn't already exist. Very odd for a DTO and not something an API consumer would expect. If this logic is only here for processing purposes, it should be in the processor.

Reply:
Done.